dialogy/plugins/text/intent_entity_mutator/__init__.py

```diff
@@ -0,0 +1,273 @@
+"""
+    IntentEntityMutatorPlugin
+"""
+
+from typing import Any, List, Optional, Dict, Iterable, Union, Callable
+
+from dialogy.base import Input, Output, Plugin, Guard
+from dialogy.types import BaseEntity, Intent
+
+from dialogy.plugins.text.intent_entity_mutator.actions_on_primitives import (
+    contain_digits,
+    is_number_absent,
+)
+
+transcript_functions_map: Dict[str, Callable[[List[str]], int]] = {
+    "is_number_absent": is_number_absent,
+    "contain_digits": contain_digits,
+}
+
+
+class IntentEntityMutatorPlugin(Plugin):
+    def __init__(
+        self,
+        dest: Optional[str] = None,
+        guards: Optional[List[Guard]] = None,
+        rules: Optional[Dict[str, List[Any]]] = None,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(guards=guards, dest=dest, **kwargs)
+
+        if rules:
+            self.rules = rules[
+                "swap_rules"
+            ]  # Rules is a python object parsed by yaml and loaded in memory
+        else:
+            raise ValueError("Rules have not been instantiated")
+
+    def check_present_absent_other_primitives(
+        self,
+        primitive: str,
+        primitive_in: Optional[List[str]],
+        primitive_not_in: Optional[List[str]],
+    ) -> bool:
+
+        if primitive and primitive_in:
+            return primitive in primitive_in
+        elif primitive and primitive_not_in:
+            return primitive not in primitive_not_in
+        return False
+
+    def check_present_absent_transcript(
+        self,
+        transcripts: Optional[Iterable[str]],
+        primitive_in: Optional[List[str]],
+        primitive_not_in: Optional[List[str]],
+    ) -> bool:
+
+        if transcripts:
+            alternatives = "".join(transcripts)
+        else:
+            raise ValueError("No transcripts provided")
+
+        if (
+            alternatives
+            and primitive_in
+            and any(word in alternatives for word in primitive_in)
+        ):
+            return True
+
+        elif (
+            alternatives
+            and primitive_not_in
+            and not any(word in alternatives for word in primitive_not_in)
+        ):
+            return True
+
+        return False
+
+    def calculate_passed_functional_conditions(
+        self,
+        transcripts: List[str],
+        conditions: Dict[str, bool],
+    ) -> int:
+
+        total_passed_conditions = 0
+        for key, val in conditions.items():
+            if transcript_functions_map[key](transcripts) == val:
+                total_passed_conditions += 1
```

Comment:
Why do you need the sum?
We discussed this that all you should care about is whether a single condition evaluates to False, then the rule shouldn't apply